pkg/skaffold/hooks/host.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/misc"
+	v1 "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest/v1"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+// hostHook represents a lifecycle hook to be executed on the host machine
+type hostHook struct {
+	cfg v1.HostHook
+	env []string // environment variables to set in the hook process
+}
+
+// run executes the lifecycle hook on the host machine
+func (h hostHook) run(ctx context.Context, out io.Writer) error {
+	if len(h.cfg.OS) > 0 && !util.StrSliceContains(h.cfg.OS, runtime.GOOS) {
+		logrus.Debugf("host hook execution skipped due to OS criteria not matched for commands:\n%s\n", strings.Join(h.cfg.Command, "\n"))
```

Comment:
This seems worthy of a warning, or at least an info (which won't be shown by default)

Reply:
agree, and maybe be a bit more specific about what the "OS criteria" is here